include/nvexec/stream/schedule_from.cuh

```diff
@@ -40,15 +47,61 @@ namespace nvexec::STDEXEC_STREAM_DETAIL_NS {
 
         template < __completion_tag Tag, class... As>
         friend void tag_invoke(Tag, __t&& self, As&&... as) noexcept {
+          using tuple_t = decayed_tuple<Tag, As...>;
+
           // As an optimization, if there are no values to persist to temporary
           // storage, skip it and simply propagate the completion signal.
           if constexpr (sizeof...(As) == 0) {
             self.operation_state_.propagate_completion_signal(Tag());
           } else {
             storage_t* storage = static_cast<storage_t*>(self.operation_state_.temp_storage_);
-            // BUGBUG TODO: construct/emplace from the device, not the host!
-            ::new (storage) storage_t();
-            storage->template emplace<decayed_tuple<Tag, As...>>(Tag(), (As&&) as...);
+            constexpr bool construct_on_device = trivially_copyable<__decay_t<As>...>;
```

Comment:
This branch could really use an introductory comment explaining what it's doing.